File: C-Abi-Bridge-Aot/Tests/CAbiBridge.Aot.Vb.Tests/TestCrypto/TestCryptoPqc/TestCryptoPqcSignature/TestCryptoPqcSignatureStateless/TestCryptoPqcSignatureMlDsa/CryptoPqcMlDsaTest.vb
Option Strict On
Option Explicit On


Namespace michele.natale.Tests
  Partial Public Module CryptoPqcMlDsaTest

    Public Sub StartNative(rounds As Int32)
      Start_Native(rounds)

      Console.WriteLine()
    End Sub

    Private Sub Start_Native(rounds As Int32)

      TestPqcMlDsaCreateKeyPairs(rounds * 10)
      TestPqcMlDsaCreateKeyPairsParam(rounds * 10)
      TestPqcMlDsaSaveLoadKeyPairs(rounds * 10)

      TestPqcMlDsaSingleSignature(rounds)
      TestPqcMlDsaSingleSignatureKpiSaveLoad(rounds)

      TestPqcMlDsaSingleSignatureFile(rounds)
      TestPqcMlDsaSingleSignatureKpiSaveLoadFile(rounds)

      TestPqcMlDsaMultiSignKpf(rounds) 'kpf = key-pair-file
      TestPqcMlDsaMultiSignFileKpf(rounds) 'kpf = key-pair-file

    End Sub

  End Module
End Namespace

File: C-Abi-Bridge-Aot/Tests/CAbiBridgeAot.Tests/TestCrypto/TestCryptoPqc/TestCryptoPqcSignature/TestCryptoPqcSignatureStateless/TestCryptoPqcSignatureMlDsa/CryptoPqcMlDsaTest.cs

namespace michele.natale.Tests;

internal sealed partial class CryptoPqcMlDsaTest
{
  public static void StartNative(int rounds)
  {
    Start_Native(rounds);

    Console.WriteLine();
  }

  private static void Start_Native(int rounds)
  {
    TestPqcMlDsaCreateKeyPairs(rounds * 10);
    TestPqcMlDsaCreateKeyPairsParam(rounds * 10);
    TestPqcMlDsaSaveLoadKeyPairs(rounds * 10);

    TestPqcMlDsaSingleSignature(rounds);
    TestPqcMlDsaSingleSignatureKpiSaveLoad(rounds);

    TestPqcMlDsaSingleSignatureFile(rounds);
    TestPqcMlDsaSingleSignatureKpiSaveLoadFile(rounds);

    TestPqcMlDsaMultiSignKpf(rounds);           //kpi = kpf = key-pair-file
    TestPqcMlDsaMultiSignFileKpfAsync(rounds);  //kpi = kpf = key-pair-file
  }
}
